src/OrchardCore.Modules/OrchardCore.Markdown/Handlers/MarkdownBodyPartHandler.cs

```diff
@@ -25,9 +26,11 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context,
         {
             return context.ForAsync<BodyAspect>(async bodyAspect =>
             {
-                if (_bodyAspect != null)
```

Comment:
This is the only line to change. It's a cached value for performance reasons I assume, but apparently this can be called for different content items. So we should store the contentitemid too and check for both value before assigning it. Otherwise just recompute it.

Reply:
> but apparently this can be called for different content items

How? With old implementation once the content item is computed the others will read the old computed value, this why we introduce the dictionary

Reply:
```
if (_bodyAspect != null && contentItem.Id == _contentItemId) 
{
  // use cache here
}

// otherwise compute a new value and cache it FOR THIS CONTENT ITEM

_bodyAspect = ... // compute body aspect for this content item
_contentItemId = contentItem.Id; // store that the cache is for this content item only
```